tests/ecosystem/upgrade/test_upgrade_ocp.py

```diff
@@ -0,0 +1,156 @@
+import logging
+import json
+import time
+
+from ocs_ci.ocs import ocp
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.framework.testlib import ManageTest
+from ocs_ci.ocs.cluster import CephCluster
+from ocs_ci.ocs.exceptions import CephHealthException
+
+logger = logging.getLogger(__name__)
+
+# TODO: add automatic image finder
+# TODO: add image type validation (ga to ga , nightly to nightly, newer than current etc.)
+# TODO: receive image as external argument
+
+
+class TestUpgradeOCP(ManageTest):
+    """
+    1. check cluster health
+    2. check OCP version
+    3. perform OCP upgrade
+    4. check all OCP ClusterOperators
+    5. check OCP version
+    5. monitor cluster health
+    """
+
+    def test_upgrade_ocp(self):
+        """
+
+        Tests OCS stability when upgrading OCP
+
+        """
+        target_image = "4.3.0-0.nightly-2020-02-17-205936"
```

Comment:
Fetch this from conf

Reply:
it will be taken from config files instead